src/rest-api/class-wp-rest-import-attachments.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+class WP_REST_Import_Attachments {
+	const NAMESPACE = 'wordpress-importer/v1';
+
+	function __construct() {
+		$this->register_routes();
+	}
+
+	protected function register_routes() {
+		register_rest_route( self::NAMESPACE, '/attachment', array(
+			// 'show_in_index' => false,
+			'methods' => WP_REST_Server::CREATABLE,
+			'callback' => array( $this, 'add_attachment' ),
+			'permission_callback' => array( $this, 'upload_and_import_permissions_check' ),
+			'args' => array(),
+		) );
+	}
+
+	function upload_and_import_permissions_check( $request ) {
+		// Begin copy from class-wp-rest-posts-controller
+		if ( ! empty( $request['id'] ) ) {
+			return new WP_Error( 'rest_post_exists', __( 'Cannot create existing post.' ), array( 'status' => 400 ) );
+		}
+
+		$post_type = get_post_type_object( 'attachment' );
+
+		if ( ! empty( $request['author'] ) && get_current_user_id() !== $request['author'] && ! current_user_can( $post_type->cap->edit_others_posts ) ) {
+			return new WP_Error( 'rest_cannot_edit_others', __( 'Sorry, you are not allowed to create posts as this user.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		if ( ! empty( $request['sticky'] ) && ! current_user_can( $post_type->cap->edit_others_posts ) ) {
+			return new WP_Error( 'rest_cannot_assign_sticky', __( 'Sorry, you are not allowed to make posts sticky.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		if ( ! current_user_can( $post_type->cap->create_posts ) ) {
+			return new WP_Error( 'rest_cannot_create', __( 'Sorry, you are not allowed to create posts as this user.' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		// Do we need this? probably not
+		// if ( ! $this->check_assign_terms_permission( $request ) ) {
+		//	return new WP_Error( 'rest_cannot_assign_term', __( 'Sorry, you are not allowed to assign the provided terms.' ), array( 'status' => rest_authorization_required_code() ) );
+		//	}
+
+		// End copy from class-wp-rest-posts-controller
+
+		if ( ! current_user_can( 'upload_files' ) ) {
+			return new WP_Error( 'upload_forbidden', __( 'Sorry, you are not allowed to upload files as this user' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		if ( ! current_user_can( 'import' ) ) {
+			return new WP_Error( 'import_forbidden', __( 'Sorry, you are not allowed to import as this user' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	function add_attachment( $request ) {
+		// Get the file via $_FILES or raw data.
+		$files = $request->get_file_params();
+		$headers = $request->get_headers();
+
+		if ( empty( $files['import'] ) ) {
+			return new WP_Error( 'rest_upload_no_data', __( 'No data supplied.' ), array( 'status' => 400 ) );
+		}
+
+		$max_bytes = apply_filters( 'import_upload_size_limit', wp_max_upload_size() );
```

Comment:
Copying [comment](https://github.com/Automattic/wordpress-importer/pull/4/files#r256161635) from #4:

---

> `$max_bytes = apply_filters( 'import_upload_size_limit', wp_max_upload_size() );`

This comes from the [code that generates the core form](https://github.com/WordPress/WordPress/blob/6ef9017c9611ca27d392a34ebbe5ae15210031fe/wp-admin/includes/template.php#L932-L956).

On my local dev env, I'm seeing this equate to `67108864` bytes (~ 67mb) which doesn't seem sufficient for _really_ big sites.

It doesn't seem like there's an explicit check on the server for this -- just the general quota application process.  Perhaps we should just pull this out of the endpoint.